memory_usage.go

```diff
@@ -20,16 +20,38 @@ type MemoryUsage struct {
 	CacheTotal uint64
 }
 
+type DBForMemoryUsage interface {
+	getDBforMemoryUsage() *C.rocksdb_t
+}
+
+func (db *DB) getDBforMemoryUsage() *C.rocksdb_t {
+	return db.c
+}
+
+func (db *TransactionDB) getDBforMemoryUsage() *C.rocksdb_t {
+	return (*C.rocksdb_t)(db.c)
```

Comment:
I'm surprised this work because is just a C-style cast to a different type, but I haven't written any C++ in years.

Reply:
Yes I was surprised too! TBH I am not certain it is perfectly safe